controllers/registry/package/list.js

```diff
@@ -15,6 +15,10 @@ var config = require('../../../config');
  * GET /:name
  */
 module.exports = function* list() {
+  // Notice, this is a workaround because of NPM bug npm/npm#9616.
+  // After npm fixed this bug, you shall delete this line.
+  common.decodeScopedPackageName(this.params);
+
   var orginalName = this.params.name || this.params[0];
```

Comment:
It is definitely **INCORRECT** to just decode `originalName`.

Here's an example, scope `@kcauchy` package `@kcauchy/pkg`.

Then because the bug npm/npm#9616, npm will send a request like this:
```
https://example-cnpm.org/@kcauchy%2fpkg
```

Since you decoded the original name and pass the check, system will now go to look for `@kcauchy%2fpkg`, but of course, there are no such scope at all.

This was my first come-up idea, but it didn't work.